util/structs_periph_gen.py

```diff
@@ -0,0 +1,78 @@
+import hjson
+import structs_gen
+
+# Path to the header_structs template
+template_path = "./sw/device/lib/drivers/template.tpl"
+
+# hjson file from which the peripherals info are taken
+mcu_cfg_file = "./mcu_cfg.hjson"
+
+# path in which the structs header files are generated, has to be formatted with
+# the name of the peripheral
+out_files_base_path = "./sw/device/lib/drivers/{}/{}_structs.h" 
+
+
+JSON_FILES = []         # list of the peripherals' json files
+OUTPUT_FILES = []       # list of the output filenames
+PERIPHERAL_NAMES = []   # list of the peripherals' names
+
+
+"""
+    Given the name of a peripheral and the path to its descriptive json file,
+    it appends the info relative to the peripheral to the three lists used
+    for generation
+"""
+def add_peripheral(name, path):
+    JSON_FILES.append(path)
+    OUTPUT_FILES.append(out_files_base_path.format(name, name))
+    PERIPHERAL_NAMES.append(name)
+
+
+"""
+    It loops over the json containing infos of several peripherals.
+    When it encounters one with the "path" field, it adds it to the 
+    lists
+"""
+def scan_peripherals(json_list):
+    for p in json_list:
+        for field in json_list[p]:
+            if(field == 'path'):
+                add_peripheral(p, json_list[p]["path"])
+
+
+
+    # for i in range(len(JSON_FILES)):
+    #     print("{}\n{}\n{}\n\n\n".format(PERIPHERAL_NAMES[i], JSON_FILES[i], OUTPUT_FILES[i]))
+
+if __name__ == "__main__":
+
+    # Open the json file adn takes the data
+    with open(mcu_cfg_file) as f:
+        data = hjson.load(f)
+
+    # Get the info relative to the various peripherals
+    ao_peripherals = data["ao_peripherals"]
+    peripherals = data["peripherals"]
+
+    # loops through the peripherals and add the useful ones
+    scan_peripherals(ao_peripherals)
+    scan_peripherals(peripherals)
+
+    # Manually add rv_plic and spi_host due to name differences between mcu_cfg.hjson and peripherals folders #
+    #rv_plic
+    JSON_FILES.append("./hw/vendor/lowrisc_opentitan/hw/ip/rv_plic/data/rv_plic.hjson")
+    OUTPUT_FILES.append("./sw/device/lib/drivers/rv_plic/rv_plic_structs.h")
+    PERIPHERAL_NAMES.append("rv_plic")
+
+    # spi_host
+    JSON_FILES.append("./hw/vendor/lowrisc_opentitan_spi_host/data/spi_host.hjson")
+    OUTPUT_FILES.append("./sw/device/lib/drivers/spi_host/spi_host_structs.h")
```

Comment:
why is the spi not taken from the hjson?

Reply:
Because in the hjson they have different names (plic and spi) with respect to the folder in which the structs need to be generated (rv_plic and spi_host). With @JoseCalero we decided to hardcode them instead of adding an additional "name" field in the hjson.

Reply:
but can't you just put the path in the path field where the json file is? I don't get what you need, the SW one can be hardwired in the script or pass the `./sw/device/lib/drivers/` as an argument, the rest is generated by the name of the peripheral no?

Reply:
Yes I can add their path to the hjson.

Reply:
Yes I can add the path to the hjson. For the output folder I have already hardcoded the base path `./sw/device/lib/drivers/` in the script, but the rest of the path cannot be generated with the name since the names are different, So I would need to hardcode them anyway